cpp/src/parquet/CMakeLists.txt

```diff
@@ -218,8 +220,9 @@ if(NOT PARQUET_MINIMAL_DEPENDENCY)
 
   # Link publicly with parquet_static (because internal users need to
   # transitively link all dependencies)
-  set(PARQUET_STATIC_LINK_LIBS ${PARQUET_STATIC_LINK_LIBS} thrift::thrift)
-endif(NOT PARQUET_MINIMAL_DEPENDENCY)
+  list(APPEND PARQUET_STATIC_LINK_LIBS thrift::thrift)
+  list(APPEND PARQUET_STATIC_INTERFACE_INSTALL_LIBS thrift::thrift)
```

Comment:
@kou Why do we need this line? `thrift::thrift` is already included in the `arrow_bundled_dependencies`. I ran into `thrift::thrift target was not found` while linking parquet_static to iceberg_static.

Reply:
Oh my bad! I didn't correctly set the `ARROW_DEPENDENCY_SOURCE` so actually thrift was found from my laptop.